File: LC_Gen_Codes_CPlusPlus/2644_no_setting.cpp
class Solution {
    public int maxDivScore(List<Integer> nums, List<Integer> divisors) {
        int maxScore = 0;
        int maxDivisor = Integer.MAX_VALUE;

        for (int divisor : divisors) {
            int score = 0;
            for (int num : nums) {
                if (num % divisor == 0) {
                    score++;
                }
            }

            if (score >= maxScore) {
                if (score > maxScore || divisor < maxDivisor) {
                    maxScore = score;
                    maxDivisor = divisor;
                }
            }
        }

        return maxDivisor;
    }
}

File: LC_Gen_Codes_Java/2644_no_setting.java
class Solution {
    public int maxDivScore(int[] nums, int[] divisors) {
        int max_score = 0;
        int max_divisor = Integer.MAX_VALUE;
        
        for (int divisor : divisors) {
            int score = 0;
            for (int num : nums) {
                if (num % divisor == 0) {
                    score++;
                }
            }
            
            if (score >= max_score) {
                if (score > max_score || divisor < max_divisor) {
                    max_score = score;
                    max_divisor = divisor;
                }
            }
        }
        
        return max_divisor;
    }
}